src/table/src/engine.rs

```diff
@@ -69,6 +71,13 @@ pub trait TableEngine: Send + Sync {
         request: CreateTableRequest,
     ) -> Result<TableRef>;
 
+    /// Returns a procedure that creates table by specific `request`.
+    async fn create_table_procedure(
```

Comment:
Is it appropriate to put `create_table_procedure` method in trait `TableEngine`? I think the procedure uses the table engine from outside, instead of integrating into it. Besides, there's `create_table` method already defined in `TableEngine`, which could cause some confusion when seeing it put side by side with `create_table_procedure`. My suggestion is that either move `create_table_procedure` out of `TableEngine`, or make `create_table` create and execute creating table procedure inside.

Reply:
Procedure is an experimental feature and I must keep the original `create_table()` method to ensure this new feature doesn't effect the old one. The final goals is to implement procedures for all DDL operations and then we could remove these old methods.

> I think the procedure uses the table engine from outside, instead of integrating into it.

The table engine itself (at least the mito engine) requires the procedure framework to handle the error/interruption during creating a table, as creating a table, especially table with multiple regions, isn't atomic. The procedure itself also needs to access the engine internals, such as table manifest, region.

> make create_table create and execute creating table procedure inside
Invoking the table engine to create a table is just a subprocedure of the whole table creation procedure. We need to return this procedure to help the parent procedure establish the relationship between them.

> move create_table_procedure out of TableEngine
Well, if this method is confusing, I could define a new trait that only provides procedure related methods.

Reply:
I move it to a new trait in [
a87f916](https://github.com/GreptimeTeam/greptimedb/pull/1035/commits/a87f916d7352f2cc4350b7a039f5840a1cb3c558)